test/services/homework/index.test.js

```diff
@@ -252,196 +278,242 @@ describe('homework service', function test() {
+				availableDate: Date.now(),
+				dueDate: '2030-11-16T12:47:00.000Z',
+				private: true,
+				archived: [],
+				lessonId: null,
+				courseId: null,
+			});
+			return {
+				teacher,
+				homework,
+			};
+		};
 
-	it('try to FIND tasks of others', async () => {
-		const user = await testObjects.createTestUser({ roles: ['teacher'] });
-		const otherUser = await testObjects.createTestUser({ roles: ['teacher'] });
-		await testObjects.createTestHomework({
-			teacherId: otherUser._id,
-			name: 'Testaufgabe',
-			description: '\u003cp\u003eAufgabenbeschreibung\u003c/p\u003e\r\n',
-			availableDate: Date.now(),
-			dueDate: '2030-11-16T12:47:00.000Z',
-			private: true,
-			archived: [otherUser._id],
-			lessonId: null,
-			courseId: null,
-		});
-		const params = await testObjects.generateRequestParamsFromUser(user);
-		params.query = {
-			teacherId: otherUser._id,
-			private: true,
+		const setupHomeworkWithGrades = async () => {
+			const [teacher, studentOne, studentTwo] = await Promise.all([
+				testObjects.createTestUser({ roles: ['teacher'] }),
+				testObjects.createTestUser({ roles: ['student'] }),
+				testObjects.createTestUser({ roles: ['student'] }),
+			]);
+			const course = await testObjects.createTestCourse({
+				teacherIds: [teacher._id],
+				userIds: [studentOne._id, studentTwo._id],
+			});
+			const homework = await testObjects.createTestHomework({
+				teacherId: teacher._id,
+				name: 'Testaufgabe',
+				description: '\u003cp\u003eAufgabenbeschreibung\u003c/p\u003e\r\n',
+				availableDate: Date.now(),
+				dueDate: '2030-11-16T12:47:00.000Z',
+				private: false,
+				archived: [teacher._id],
+				lessonId: null,
+				courseId: course._id,
+			});
+			const submission = await testObjects.createTestSubmission({
+				schoolId: course.schoolId,
+				courseId: course._id,
+				homeworkId: homework._id,
+				studentId: studentOne._id,
+				comment: 'hello teacher, my dog has eaten this database entry...',
+				grade: 67,
+			});
+			return {
+				teacher,
+				students: [studentOne, studentTwo],
+				course,
+				homework,
+				submission,
+			};
 		};
-		const result = await homeworkService.find(params);
-		expect(result.total).to.equal(0);
-	});
 
-	const setupHomeworkWithGrades = async () => {
-		const [teacher, studentOne, studentTwo] = await Promise.all([
-			testObjects.createTestUser({ roles: ['teacher'] }),
-			testObjects.createTestUser({ roles: ['student'] }),
-			testObjects.createTestUser({ roles: ['student'] }),
-		]);
-		const course = await testObjects.createTestCourse({
-			teacherIds: [teacher._id],
-			userIds: [studentOne._id, studentTwo._id],
+		it('I am be able to FIND my own tasks', async () => {
+			const { teacher } = await setUpPrivateHomework();
+			const params = await testObjects.generateRequestParamsFromUser(teacher);
+			params.query = {};
+			const result = await homeworkService.find(params);
+			expect(result.total).to.be.above(0);
```

Comment:
maybe it will be better to do an exact match here to be sure you get no more you expacted